### DISCUSSION TITLE
kernel to find "stars" marks

### DISCUSSION THREAD

**Discussion by NcMichel** [ImageMagick/ImageMagick#3239] [Help]
Hello, 
working on poll scans, I need some fiducial marks to resize and `-distort Perspective` the resulting scans.

I try to isolate some "stars" shapes as those marks
![image](https://user-images.githubusercontent.com/56865796/107846593-ddd99d80-6e1f-11eb-933a-bbea35f04a15.png)
using various kernels, but my results are really poor.

The best way I found is 
`-morphology close 'cross:2x2;plus:4x4'`
but this find other shapes as well, like 
![image](https://user-images.githubusercontent.com/56865796/107846658-56d8f500-6e20-11eb-850e-e7dd47f445fc.png) or 
![image](https://user-images.githubusercontent.com/56865796/107846665-67896b00-6e20-11eb-94fe-c37d29b09a76.png)

I did try with `-morphology HMT LineJunctions` which give same results....

Any idea about a better kernel or a better method greatly appreciated.

Merci
Michel

**Comment by snibgo** [comment]
What varies between the scans? Variation might include: scale and/or rotation and/or translation. The marks should be designed to be unvarying under whichever variation occurs. For example, if you have rotation and translation but no scaling, then circular marks are appropriate, but marks like "+" or "x" are not appropriate because they change when rotated, so searching for them is harder.

**Reply by NcMichel** [reply]
Thanks for advice.
The overall polling page looks like this:
![blank_survey1500](https://user-images.githubusercontent.com/56865796/107848321-83473e00-6e2d-11eb-8c17-e598d426d199.png)

The scan can be rotated, distort, reversed. and of course translated ... the pollings are written by teenagers, then flat-scanned.

The scans are previously "deskewed".

If I use, for instance, circles, my problem will become
"how to find a circle (maybe, distort by scan) and nothing else in the picture".

As you may notice, fiducial marks are positioned so it's not too difficult, once you have marks positions, to figure if the scan is rotated or flipped, and readjust as needed. It did work in my tests.

However, if there is a better way, Ill be more than happy to learn from you.

Thanks for your care
Michel
